projects/core/src/checkout/http-interceptors/checkout-cart.interceptor.spec.ts

```diff
@@ -0,0 +1,206 @@
+import { HttpClient, HTTP_INTERCEPTORS } from '@angular/common/http';
+import {
+  HttpClientTestingModule,
+  HttpTestingController,
+} from '@angular/common/http/testing';
+import { TestBed } from '@angular/core/testing';
+import { of, throwError } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+import { MultiCartService } from '../../cart';
+import { RoutingService } from '../../routing';
+import { CheckoutCartInterceptor } from './checkout-cart.interceptor';
+
+class MockRoutingService {
+  go() {}
+
+  getRouterState() {
+    return of();
+  }
+}
+
+class MultiCartServiceStub {
+  reloadCart() {}
+}
+
+describe('CheckoutCartInterceptor', () => {
+  let httpMock: HttpTestingController;
+  let http: HttpClient;
+  let routingService: RoutingService;
+  let multiCartService: MultiCartService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [
+        {
+          provide: HTTP_INTERCEPTORS,
+          useClass: CheckoutCartInterceptor,
+          multi: true,
+        },
+        { provide: RoutingService, useClass: MockRoutingService },
+        { provide: MultiCartService, useClass: MultiCartServiceStub },
+      ],
+    });
+
+    httpMock = TestBed.inject(HttpTestingController);
+    http = TestBed.inject(HttpClient);
+    routingService = TestBed.inject(RoutingService);
+    multiCartService = TestBed.inject(MultiCartService);
+
+    spyOn(routingService, 'go');
+    spyOn(multiCartService, 'reloadCart');
+  });
+
+  describe('intercept', () => {
+    it('should redirect to cart page if the cart is not found', () => {
+      spyOn(routingService, 'getRouterState').and.returnValue(
+        of({ state: { semanticRoute: 'checkoutDelivery' } } as any)
+      );
+
+      http
```

Comment:
This one can be extracted to a constant or generated by a function to avoid duplication